src/rpc/core_rpc_server.cpp

```diff
@@ -3324,5 +3339,84 @@ namespace cryptonote
     res.status = CORE_RPC_STATUS_OK;
     return true;
   }
+  //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_get_lns_names_to_owners(const COMMAND_RPC_GET_LNS_NAMES_TO_OWNERS::request &req, COMMAND_RPC_GET_LNS_NAMES_TO_OWNERS::response &res, epee::json_rpc::error &error_resp, const connection_context *ctx)
+  {
+    if (exceeds_quantity_limit(ctx, error_resp, m_restricted, req.entries.size(), COMMAND_RPC_GET_LNS_NAMES_TO_OWNERS::MAX_REQUEST_ENTRIES))
+      return false;
+
+    lns::name_system_db const &db = m_core.get_blockchain_storage().name_system_db();
+    for (size_t request_index = 0; request_index < req.entries.size(); request_index++)
+    {
+      COMMAND_RPC_GET_LNS_NAMES_TO_OWNERS::request_entry const &request = req.entries[request_index];
+      if (exceeds_quantity_limit(ctx, error_resp, m_restricted, request.types.size(), COMMAND_RPC_GET_LNS_NAMES_TO_OWNERS::MAX_TYPE_REQUEST_ENTRIES, "types"))
+        return false;
+
+      for (uint16_t type : request.types)
+      {
+        if (lns::mapping_record mapping = db.get_mapping(type, request.name))
+        {
+          if (lns::user_record user = db.get_user_by_id(mapping.user_id))
```

Comment:
Updated, not that well versed, but did you mean like this?

```c++
  char constexpr GET_MAPPING_BY_NAME_AND_TYPE_SQL[] = R"(SELECT * FROM "mappings" JOIN "user" ON "mappings"."user_id" = "user"."id" WHERE "type" = ? AND "name" = ?)";
```

Reply:
I actually meant building a string with the requisite number of `?,`s in it, then using an `IN` clause and binding the right number of values.

Basically: `column IN ('a', 'b', 'c')` is equivalent to `(column == 'a' OR column == 'b' OR column == 'c')`, though often a bit more efficient, and so, unless I'm misreading the logic, it should be possible to fetch everything all in one go here.  But it will require building the string dynamically to have the right number of placeholders in the `IN` clause (and then preparing and executing on the fly, rather than storing a pre-prepared statement).

Reply:
I was misreading this earlier, and see there's only one `name` here, so `name` doesn't need it; just:
```SQL
SELECT * FROM "mappings" JOIN "user" ON "mappings"."user_id" = "user"."id" where "type" IN (?, ?, ?) AND "name" = ?
```
where the `(?, ?, ?)` part of the query is generated on the fly (and arguments are similarly bound on the fly according to the size of `request.types`.

Reply:
Yeah makes sense, I've updated to build the query at run time